lib/utils.js

```diff
@@ -7,17 +7,25 @@ var assert = require("assert");
  * @returns Configuration Object
  */
 exports.parseMongoConnectionURL = function (mongoConnectionURL) {
-  var result = {};
+  var result = {
+    database: {
+      driver_options: {}
+    }
+  };
 
   //The connection string will be of the form mongodb://user:password@host:port,host2port2/databasename?option=someOption
   //String to break down the mongodb url into a ditch config hash
   var auth_hosts_path_options = new Buffer(mongoConnectionURL).toString().split("//")[1];
-
-  //user:password , host:port,host2:port2/databasename?option=someOption
-  var auth = auth_hosts_path_options.split("@")[0];
-
-
-  var hosts_path_options = auth_hosts_path_options.split("@")[1];
+  var hosts_path_options = auth_hosts_path_options;
+  var auth = null;
+
+  // If auth is included, split that from the rest of the string
+  // and configure the auth config
+  if (auth_hosts_path_options.indexOf("@") > -1) {
+    //user:password , host:port,host2:port2/databasename?option=someOption
+    var auth = auth_hosts_path_options.split("@")[0];
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "auth" as this name is already used in declaration at line 20. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2814)